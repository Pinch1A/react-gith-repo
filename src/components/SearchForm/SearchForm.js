import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  background-color: ${(props) => (props.primary ? 'orange' : 'grey')};
  color: white;
  border: none;
  width: 30%;
  border-radius: 5px;
`
const FormWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 30px;
`
const SearchInput = styled.input`
  min-width: 300px;
`

const StyledForm = styled.form`
  display: flex;
`

const SearchForm = ({ searchString, handleForm, onChange, clearSearch }) => (
  <FormWrapper>
    <StyledForm autoComplete="off" onSubmit={() => handleForm}>
      <SearchInput
        name="repoName"
        onChange={(e) => onChange(e.target.value)}
        placeholder="Search Repositories by name"
        value={searchString}
      />
    </StyledForm>

    <Button onClick={clearSearch}>Clear Results</Button>
  </FormWrapper>
)
export default SearchForm
