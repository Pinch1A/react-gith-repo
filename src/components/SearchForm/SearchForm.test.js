import React from 'react'
import SearchForm from './SearchForm'
import { shallow } from 'enzyme'

const props = {
  searchString: '',
  handleForm: jest.fn(),
  onChange: jest.fn(),
  clearSearch: jest.fn(),
}

describe('<SearchForm>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SearchForm {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
