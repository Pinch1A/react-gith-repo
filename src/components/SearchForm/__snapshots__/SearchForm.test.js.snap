// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm> renders without crashing 1`] = `
<styled.div>
  <styled.form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <styled.input
      name="repoName"
      onChange={[Function]}
      placeholder="Search Repositories by name"
      value=""
    />
  </styled.form>
  <styled.button
    onClick={[MockFunction]}
  >
    Clear Results
  </styled.button>
</styled.div>
`;
